compiler/src/dotty/tools/dotc/transform/FirstTransform.scala

```diff
@@ -217,6 +217,27 @@ class FirstTransform extends MiniPhaseTransform with InfoTransformer with Annota
       case _ => tree
     }
 
+  /** Perform one of the following simplification if applicable:
+   *
+   *      true  && y   ==>  y
+   *      false && y   ==>  false
+   *      true  || y   ==>  true
+   *      false || y   ==>  y
+   */
+  private def foldCondition(tree: Apply)(implicit ctx: Context) = tree.fun match {
+    case Select(x @ Literal(Constant(c: Boolean)), op) =>
+      tree.args match {
+        case y :: Nil if y.tpe.widen.isRef(defn.BooleanClass) =>
+          op match {
+            case nme.ZAND => if (c) y else x
+            case nme.ZOR  => if (c) x else y
```

Comment:
Probably doesn't matter in this case, but I prefer to check the symbol's name (or identity), rather than the selected name, so as to avoid mis-classification in the presence of renaming imports.